config/config-api/src/main/java/com/thoughtworks/go/util/command/UrlArgument.java

```diff
@@ -35,26 +36,42 @@ public UrlArgument(String url) {
         secretParams = SecretParams.parse(this.url);
     }
 
-    public String forCommandline() {
+    @Override
+    public boolean hasSecretParams() {
+        return secretParams != null && !secretParams.isEmpty();
+    }
+
+    @Override
+    public SecretParams getSecretParams() {
+        return secretParams;
+    }
+
+    @Override
+    public String originalArgument() {
         return url;
     }
 
+    @Override
     public String forDisplay() {
+        if (hasSecretParams()) {
```

Comment:
We might have to change this, e.g `https://#{SECRET[id][key]}:password@foo.com/bar.git` how will this be handled?